packages/ember/tests/routing/query_params_test.js

```diff
@@ -3351,6 +3351,40 @@ if (isEnabled('ember-routing-route-configured-query-params')) {
       promiseResolve();
     });
   });
+
+  QUnit.test('undefined queryParams is treated as empty object', function() {
+    expect(1);
+
+    App.Router.map(function() {
+      this.route('foo');
+      this.route('bar');
+    });
+
+    App.FooRoute = Route.extend({
+      beforeModel() {
+        this.transitionTo('bar', { queryParams: undefined });
```

Comment:
Does this error only occur when explicitly passing `undefined`?  Would the following also fail:

```js
this.transitionTo('bar');

// or
this.transitionTo('bar',  {});
```

?